src/cdk-experimental/tree/tree.ts

```diff
@@ -147,10 +154,10 @@ export class CdkTree<T> implements CollectionViewer, OnInit, OnDestroy {
    * render change subscription if one exists. If the data source is null, interpret this by
    * clearing the node outlet. Otherwise start listening for new data.
    */
-  private _switchDataSource(dataSource: DataSource<T>) {
+  private _switchDataSource(dataSource: DataSource<T> | Observable<T[]> | T[]) {
     this._data = new Array<T>();
 
-    if (this.dataSource) {
+    if (this.dataSource instanceof DataSource) {
```

Comment:
We can into problems with the table using `instanceof` for this since people often create fakes for tests that are just object literals with a `connect` method. @andrewseguin what did you end up doing?

Reply:
Updated. Using the same way as table now. Please take a look. Thanks!